contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/util/SimpleHttp.java

```diff
@@ -0,0 +1,235 @@
+import java.io.InputStream;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+
+/**
+ * This class performs the actual HTTP requests for the HTTP Storage Plugin. The core method is the getInputStream()
+ * method which accepts a url and opens an InputStream with that URL's contents.
+ */
+public class SimpleHttp {
+  private static final Logger logger = LoggerFactory.getLogger(SimpleHttp.class);
+
+  private final OkHttpClient client;
+
+  private final HttpStoragePluginConfig config;
+
+  private final FragmentContext context;
+
+  private final HttpAPIConfig apiConfig;
+
+  public SimpleHttp(HttpStoragePluginConfig config, FragmentContext context, String connectionName) {
+    this.config = config;
+    this.context = context;
+    this.apiConfig = config.connections().get(connectionName);
+    client = setupServer();
+  }
+
+
+
+  public InputStream getInputStream(String urlStr) {
+    Request.Builder requestBuilder;
+
+    // The configuration does not allow for any other request types other than POST and GET.
+    if (apiConfig.method().equals("get")) {
+      // Handle GET requests
+      requestBuilder = new Request.Builder().url(urlStr);
+    } else {
+      // Handle POST requests
+      FormBody.Builder formBodyBuilder = buildPostBody();
+      requestBuilder = new Request.Builder()
+        .url(urlStr)
+        .post(formBodyBuilder.build());
+    }
+
+    // Add headers to request
+    if (apiConfig.headers() != null) {
+      for (Map.Entry<String, String> entry : apiConfig.headers().entrySet()) {
+        String key = entry.getKey();
+        String value = entry.getValue();
+        requestBuilder.addHeader(key, value);
+      }
+    }
+
+    // Build the request object
+    Request request = requestBuilder.build();
+    logger.debug("Headers: {}", request.headers());
+
+    try {
+      // Execute the request
+      Response response = client
+        .newCall(request)
+        .execute();
+      logger.debug(response.toString());
+
+      // If the request is unsuccessful, throw a UserException
+      if (!response.isSuccessful()) {
+        throw UserException.dataReadError()
+          .message("Error retrieving data: " + response.code() + " " + response.message())
+          .addContext("Response code: ", response.code())
+          .build(logger);
+      }
+      logger.debug("HTTP Request for {} successful.", urlStr);
+      logger.debug("Response Headers: {} ", response.headers().toString());
+
+      // Return the InputStream of the response
+      return response.body().byteStream();
+    } catch (IOException e) {
+      throw UserException.functionError()
+        .message("Error retrieving data")
+        .addContext(e.getMessage())
+        .build(logger);
+    }
+  }
+
+  /**
+   * Function configures the OkHTTP3 server object with configuration info from the user.
+   *
+   * @return OkHttpClient configured server
+   */
+  private OkHttpClient setupServer() {

```

Comment:
This isn't really a "server", it is a "client" to a server.

Reply:
Renamed this `setupHttpClient`